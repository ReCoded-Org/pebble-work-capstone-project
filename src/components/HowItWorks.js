import { config } from "@fortawesome/fontawesome-svg-core";
import Image from "next/image";
import Link from "next/link";

import "@fortawesome/fontawesome-svg-core/styles.css";
config.autoAddCss = false;
import { faChevronRight } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

function HowItWorks() {
    return (
        <div className='how-it-works'>
            <h1 className='mt-5 flex justify-center text-4xl font-bold md:m-3 md:text-4xl lg:mb-9 lg:text-4xl'>
                How It Works
            </h1>
            <div className='flex flex-col items-center justify-center lg:flex-row'>
                <div className='m-4 w-64 overflow-hidden rounded pb-6 md:m-4 md:w-2/3 md:pb-0 lg:m-8 lg:w-80'>
                    <Image
                        class='w-full'
                        src='/images/register.png'
                        alt='Register'
                        width='100%'
                        height='100%'
                        layout='responsive'
                    />
                    <div className='px-6 py-4'>
                        <div className='m-2 flex justify-center text-center text-2xl font-bold md:text-3xl lg:text-lg'>
                            Register
                        </div>
                        <p className='text-center text-gray-700 md:text-2xl lg:text-lg'>
                            Body placeholder for text paragraph, a paragraph is
                            a self-contained unit of text description.
                        </p>
                    </div>
                    <div className='mb-4 p-2 px-6 pt-4'>
                        <h1 className='title text-center font-bold md:text-2xl lg:text-lg'>
                            {" "}
                            <Link href='/posts/first-post'>
                                <a>
                                    Sign up{" "}
                                    <FontAwesomeIcon icon={faChevronRight} />
                                </a>
                            </Link>
                        </h1>
                    </div>
                </div>
                <div className='m-4 w-64 overflow-hidden rounded pb-6 md:m-4 md:w-2/3 md:pb-0 lg:m-8 lg:w-80'>
                    <Image
                        class='w-full'
                        src='/images/attend.png'
                        alt='Attend'
                        width='100%'
                        height='100%'
                        layout='responsive'
                    />
                    <div className='px-6 py-4'>
                        <div className='m-2 flex justify-center text-center text-2xl font-bold md:text-3xl lg:text-lg'>
                            Attend events
                        </div>
                        <p className='text-center text-gray-700 md:text-2xl lg:text-lg'>
                            Body placeholder for text paragraph, a paragraph is
                            a self-contained unit of text description.
                        </p>
                    </div>
                    <div className='mb-4 p-2 px-6 pt-4'>
                        <h1 className='title text-center font-bold md:text-2xl lg:text-lg'>
                            {" "}
                            <Link href='/posts/first-post'>
                                <a>
                                    View events{" "}
                                    <FontAwesomeIcon icon={faChevronRight} />
                                </a>
                            </Link>
                        </h1>
                    </div>
                </div>
                <div className='m-4 w-64 overflow-hidden rounded pb-6 md:m-4 md:w-2/3 md:pb-0 lg:m-8 lg:w-80'>
                    <Image
                        class='w-full'
                        src='/images/organize.png'
                        alt='Organize'
                        width='100%'
                        height='100%'
                        layout='responsive'
                    />
                    <div className='px-6 py-4'>
                        <div className='m-2 flex justify-center text-center text-2xl font-bold md:text-3xl lg:text-lg'>
                            Organize on your own!
                        </div>
                        <p className='text-center text-gray-700 md:text-2xl lg:text-lg'>
                            Body placeholder for text paragraph, a paragraph is
                            a self-contained unit of text description.
                        </p>
                    </div>
                    <div className='mb-4 p-2 px-6 pt-4'>
                        <h1 className='title text-center font-bold md:text-2xl lg:text-lg'>
                            {" "}
                            <Link href='/posts/first-post'>
                                <a>
                                    Get started{" "}
                                    <FontAwesomeIcon icon={faChevronRight} />
                                </a>
                            </Link>
                        </h1>
                    </div>
                </div>
            </div>
        </div>
    );
}
export default HowItWorks;
