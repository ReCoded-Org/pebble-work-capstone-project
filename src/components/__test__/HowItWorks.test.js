import renderer from "react-test-renderer";

import HowItWorks from "../HowItWorks";

it("renders correctly", () => {
    const tree = renderer.create(<HowItWorks />).toJSON();
    expect(tree).toMatchSnapshot();
});
