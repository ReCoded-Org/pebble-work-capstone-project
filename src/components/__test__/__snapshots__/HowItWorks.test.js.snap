// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="how-it-works"
  id="how-it-works"
>
  <h1
    className="mt-5 flex justify-center text-xl font-bold md:m-3 md:text-4xl lg:mb-9 lg:text-4xl"
  >
    landingPage.howItWorks.title
  </h1>
  <div
    className="flex flex-col items-center justify-center lg:flex-row"
  >
    <div
      className="m-4 w-64 overflow-hidden rounded pb-6 md:m-4 md:w-2/3 md:pb-0 lg:m-8 lg:w-80"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "opacity": 1,
              "padding": 0,
              "paddingTop": "100%",
              "width": "initial",
            }
          }
        />
        <img
          alt="Register"
          class="w-full"
          data-nimg="responsive"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Register"
            class="w-full"
            data-nimg="responsive"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fregister.png&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fregister.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fregister.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fregister.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fregister.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fregister.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fregister.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fregister.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fregister.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <div
        className="px-6 py-4"
      >
        <div
          className="m-2 flex justify-center text-center font-bold md:text-3xl lg:text-lg"
        >
          landingPage.howItWorks.register
        </div>
        <p
          className="text-center text-sm text-gray-700 md:text-xl lg:text-lg"
        >
          landingPage.howItWorks.registerDescription
        </p>
      </div>
      <div
        className="mb-4 p-2 px-6 pt-4"
      >
        <h1
          className="title text-center font-bold md:text-2xl lg:text-lg"
        >
           
          <a
            href="/signup"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            landingPage.howItWorks.signUp
             
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 480c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L242.8 256L73.38 86.63c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25l-192 192C112.4 476.9 104.2 480 96 480z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </h1>
      </div>
    </div>
    <div
      className="m-4 w-64 overflow-hidden rounded pb-6 md:m-4 md:w-2/3 md:pb-0 lg:m-8 lg:w-80"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "opacity": 1,
              "padding": 0,
              "paddingTop": "100%",
              "width": "initial",
            }
          }
        />
        <img
          alt="Attend"
          class="w-full"
          data-nimg="responsive"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Attend"
            class="w-full"
            data-nimg="responsive"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fattend.png&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fattend.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fattend.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fattend.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fattend.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fattend.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fattend.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fattend.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fattend.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <div
        className="px-6 py-4"
      >
        <div
          className="m-2 flex justify-center text-center font-bold md:text-3xl lg:text-lg"
        >
          landingPage.howItWorks.attendEvents
        </div>
        <p
          className="text-center text-sm text-gray-700 md:text-xl lg:text-lg"
        >
          landingPage.howItWorks.attendEventsDescription
        </p>
      </div>
      <div
        className="mb-4 p-2 px-6 pt-4"
      >
        <h1
          className="title text-center font-bold md:text-2xl lg:text-lg"
        >
           
          <a
            href="/events"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            landingPage.howItWorks.exploreEvents
             
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 480c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L242.8 256L73.38 86.63c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25l-192 192C112.4 476.9 104.2 480 96 480z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </h1>
      </div>
    </div>
    <div
      className="m-4 w-64 overflow-hidden rounded pb-6 md:m-4 md:w-2/3 md:pb-0 lg:m-8 lg:w-80"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "opacity": 1,
              "padding": 0,
              "paddingTop": "100%",
              "width": "initial",
            }
          }
        />
        <img
          alt="Organize"
          class="w-full"
          data-nimg="responsive"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="Organize"
            class="w-full"
            data-nimg="responsive"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Forganize.png&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Forganize.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Forganize.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Forganize.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Forganize.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Forganize.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Forganize.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Forganize.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Forganize.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
      <div
        className="px-6 py-4"
      >
        <div
          className="m-2 flex justify-center text-center font-bold md:text-3xl lg:text-lg"
        >
          landingPage.howItWorks.organizeYourOwn
        </div>
        <p
          className="text-center text-sm text-gray-700 md:text-xl lg:text-lg"
        >
          landingPage.howItWorks.organizeYourOwnDescription
        </p>
      </div>
      <div
        className="mb-4 p-2 px-6 pt-4"
      >
        <h1
          className="title text-center font-bold md:text-2xl lg:text-lg"
        >
           
          <a
            href="/signup"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            landingPage.howItWorks.createEvents
             
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-chevron-right "
              data-icon="chevron-right"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 480c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L242.8 256L73.38 86.63c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l192 192c12.5 12.5 12.5 32.75 0 45.25l-192 192C112.4 476.9 104.2 480 96 480z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </h1>
      </div>
    </div>
  </div>
</div>
`;
